Model/Tax/Sales/Total/Quote/Tax.php

```diff
@@ -328,6 +344,23 @@ public function mapItem(
             }
         }
 
+        try {
+            $product = $this->productRepository->getById($item->getProductId(), false, $item->getStore());
+
+            // Configurable products should use the PTC of the child (when available)
+            if ($product->getTypeId() == 'configurable') {
+                $children = $item->getChildren();
+
+                if (is_array($children) && isset($children[0])) {
+                    $product = $this->productRepository->getById($children[0]->getProductId(), false, $item->getStore());
+                }
+            }
+
+            $extensionAttributes->setTjPtc($product->getTjPtc());
+        } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
+            $this->logger->log($e->getMessage());
```

Comment:
What message would be returned here when the entity does not exist? Would it be helpful enough for the merchant to debug the issue and understand which product doesn't exist? Is it worth logging or better as a no-op?

Reply:
The default error message is "Product does not exist".  We've seen similar errors in the past where products are deleted but still referenced.  

We can prefix it with a message of our own, but some error should appear in our logs to aid in debugging efforts.

Reply:
@ryanreeves-taxjar I think it would be helpful if there was context around what specific line item has a product that does not exist. The deleted product ID may not be helpful since it's already deleted, but maybe the line ID / index and the line amount?

```
Product does not exist in line item `$itemDataObject->getCode()` for $`$item->getRowTotal()`
```